plotly/dashboard_objs/dashboard_objs.py

```diff
@@ -0,0 +1,302 @@
+"""
+dashboard_objs
+==========
+
+A module which is used to create dashboard objects, manipulate them and then
+upload them.
+
+"""
+
+import copy
+import json
+import requests
+import pprint
+import webbrowser
+
+import plotly
+
+from plotly import exceptions
+from plotly.utils import node_generator
+from plotly.api.v2.utils import build_url
+
+username = plotly.tools.get_credentials_file()['username']
+api_key = plotly.tools.get_credentials_file()['api_key']
+headers = {'Plotly-Client-Platform': 'nteract'}
+
+
+class EmptyBox(dict):
+    def __init__(self):
+        self['type'] = 'box'
+        self['boxType'] = 'empty'
+
+
+class Box(dict):
+    def __init__(self, fileId='', shareKey=None, title=''):
+        self['type'] = 'box'
+        self['boxType'] = 'plot'
+        self['fileId'] = fileId
+        self['shareKey'] = shareKey
+        self['title'] = title
+
+
+class Container(dict):
+    def __init__(self, box_1=EmptyBox(), box_2=EmptyBox(), size=400,
+                 sizeUnit='px', direction='vertical'):
+        self['type'] = 'split'
+        self['size'] = size
+        self['sizeUnit'] = sizeUnit
+        self['direction'] = direction
+        self['first'] = box_1
+        self['second'] = box_2
+
+
+class Dashboard(dict):
+    def __init__(self, dashboard_json=None, backgroundColor='#FFFFFF',
+                 boxBackgroundColor='#ffffff', boxBorderColor='#d8d8d8',
+                 boxHeaderBackgroundColor='#f8f8f8', foregroundColor='#333333',
+                 headerBackgroundColor='#2E3A46', headerForegroundColor='#FFFFFF',
+                 links=[], logoUrl='', title='Untitled Dashboard'):
```

Comment:
^^ what do you think about this? Attempting to add too many frills has been a problem for us in the past. I don't want to repeat old mistakes if there's no reason to.

The important thing to note here is that the backend API should remain the ultimate source of truth. Dashboards are small JSON blobs; let's just throw whatever the user puts in there at our API and report back success/failure.